sksurv/util.py

```diff
@@ -260,3 +260,83 @@ def safe_concat(objs, *args, **kwargs):
         concatenated[name] = pd.Categorical(concatenated[name], **params)
 
     return concatenated
+
+
+class _ConditionalAvailableProperty:
```

Comment:
I wonder if you could inherit from the built-in `property`, because it is a class. This could allow removing the setter/getter/deleter code.